pkg/linode-bs/controllerserver.go

```diff
@@ -265,11 +266,17 @@ func (linodeCS *LinodeControllerServer) ControllerPublishVolume(ctx context.Cont
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
-	opts := &linodego.VolumeAttachOptions{
-		LinodeID: linodeID,
-		ConfigID: 0,
+	// Check to see if there is room to attach this volume to the instance.
+	if canAttach, err := linodeCS.canAttach(ctx, linodeID); err != nil {
+		return &csi.ControllerPublishVolumeResponse{}, status.Error(codes.Internal, err.Error())
+	} else if !canAttach {
+		return &csi.ControllerPublishVolumeResponse{}, status.Error(codes.ResourceExhausted, "max number of volumes already attached to instance")
```

Comment:
should we add the max number of volumes that can be attached as an indicator to the user?

Reply:
Sure thing!